### PR TITLE
Update PNSE for .NET 6

### LINKED ISSUE #1683: Add PlatformNotSupported exception for APIs/platforms not supported in .NET Core
APIs might have a different behavior depending on the platform in .NET Core and might throw the PlatformNotSupported exception. We need to call that out in the Exception section of the APIs impacted.

I remember seeing spreadsheets flowing around with that information @terrajobst. Do we have the latest version of that @terrajobst?

/cc @bartonjs

### LINKED ISSUE #6666: Add new PNSEs for .NET 6
Some new PNSEs have been added in .NET 6, for example, for all the StrongNameKeyPair members. The API docs should be updated accordingly. Related to https://github.com/dotnet/docs/issues/23916.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Reflection/StrongNameKeyPair.cs#L9: now all constructors and APIs will throw.

_Originally posted by @buyaa-n in https://github.com/dotnet/dotnet-api-docs/issues/6656#issuecomment-827019893_

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #6666, fixes #1683.

- Add missing exceptions. 
- Standardize wording.
- Update link text for an article.

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
System.Runtime.InteropServices and MarshalByObject LGTM.

**Comment by bartonjs** [comment]
System.Security.* LGTM

File: xml/System.Reflection/StrongNameKeyPair.xml [hunk 1]
```diff
@@ -130,6 +130,7 @@
         <exception cref="T:System.ArgumentNullException">
```
Comment (buyaa-n) [inline xml/System.Reflection/StrongNameKeyPair.xml:130]: This type is Obsoleted, not sure if we need to add anything about that

**Review by buyaa-n — APPROVED** [review]
Thanks! System.Reflection related changes LGTM

**Comment by kouvel** [comment]
System.Threading changes LGTM

File: xml/System.Reflection/StrongNameKeyPair.xml (hunk 1 above)
Comment (gewarren) [reply xml/System.Reflection/StrongNameKeyPair.xml:130]: Once we update the APIs for preview 4, that info should get imported (it's not something we add manually).

**Review by adamsitnik — APPROVED** [review]
System.IO and System.Diagnostics.Process LGTM, thank you @gewarren

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

You can :shipit: when ready.

**Comment by gewarren** [comment]
Thank you all for the reviews!

**Pull request merged**